app.rb

```diff
@@ -0,0 +1,27 @@
+require 'sinatra'
+require 'jwt'
+
+METABASE_EMBEDDING_SECRET = ENV.fetch('METABASE_EMBEDDING_SECRET')
+METABASE_SITE_URL = ENV.fetch('METABASE_SITE_URL') || 'http://localhost:3000'
+METABASE_EMBED_DASHBOARD_ID= ENV.fetch('METABASE_EMBED_DASHBOARD_ID')
+
+get '/' do
+  payload = {
+    :resource => {:dashboard => 8},
+    :params => {          
+    },
+    :exp => Time.now.to_i + (60 * 10) # 10 minute expiration
+  }
+  token = JWT.encode payload, METABASE_EMBEDDING_SECRET
+  iframe_url = METABASE_SITE_URL + "/embed/dashboard/" + token + "#bordered=true&titled=true"
```

Comment:
I'm fifty fifty on whether to use string interpolation or concatenation .

Should the `#` be a `?`? Not really sure what happens to the `titled` param.  If it comes after an anchor.

Reply:
>I'm fifty fifty on whether to use string interpolation or concatenation .

To be fair, I am not a fan of neither of the two. I think any of the two ways to construct the URL is brittle.
For instance, if there is a trailing `/` in `METABASE_SITE_URL`, then there will be a double `//` in the URL.

I will explore using the URI module to construct the URL instead.